src/main/cpp/_nix_based/jssc.cpp

```diff
@@ -26,6 +26,7 @@
 #include <limits.h>
 #include <stdio.h>
 #include <fcntl.h>
+#include <new> // std::bad_alloc
```

Comment:
I have run this on a branch with all the new PRs merged, and this does not work on macos pipeline:
https://github.com/pietrygamat/jssc/actions/runs/14685432138/job/41213415862

```
cmake-build:
     [exec] [ 50%] Building CXX object CMakeFiles/jssc.dir/src/main/cpp/_nix_based/jssc.cpp.o
     [exec] /Users/runner/work/jssc/jssc/src/main/cpp/_nix_based/jssc.cpp:29:10: fatal error: 'new' file not found
     [exec] #include <new> // std::bad_alloc
     [exec]          ^~~~~
     [exec] 1 error generated.
     [exec] make[2]: *** [CMakeFiles/jssc.dir/src/main/cpp/_nix_based/jssc.cpp.o] Error 1
     [exec] make[1]: *** [CMakeFiles/jssc.dir/all] Error 2
     [exec] make: *** [all] Error 2
```

Reply:
@pietrygamat I have a feeling this is due to GitHub switching their `macos-latest` runners from Intel to Apple Silicon exposing something obscurely broke with the SDK detection.  Since I have Apple Silicon to test on, I'll do some testing locally to see what's going on.

Reply:
Hmm... I was starting to scratch my head and then remembered we've improved some of the CI logic since this PR was created.  I'm going to fast-forward this PR just to be on the safe side.

Reply:
> Hmm... I was starting to scratch my head and then remembered we've improved some of the CI logic since this PR was created. I'm going to fast-forward this PR just to be on the safe side.

Oh, I forgot that @hiddenalpha doesn't allow pushes to his fork, so he'll have to do the fast-forward or we'll have to open a new PR to test this out.

Reply:
> forgot that @hiddenalpha doesn't allow pushes to his fork

Oh. Was not aware this little tick prevents further commits. Do you have access now?

Reply:
So I can reproduce this and ignoring the fast-forward, I still can't find a way to resolve this locally... Here's a thread that describes the problem, but sadly most of apple's forums use XCode which isn't necessarily helpful for cmake...  

https://developer.apple.com/forums/thread/106114
https://developer.apple.com/forums/thread/108537

For other C++ projects I've added `-stdlib=libc++` to the CMAKE_CXX_FLAGS, but IIRC that hasn't been needed for quite some time (and my tests with jssc didn't work).   I'm still investigating this...

Reply:
> > forgot that @hiddenalpha doesn't allow pushes to his fork
> 
> Oh. Was not aware this little tick prevents further commits. Do you have access now?

Thank you, I'll test it out soon!

Reply:
> Do you have access now?

Still shows access denied unfortunately.

Reply:
> > I'm still investigating this...
> 
> So it appears this `<new>` missing is caused by the `include/cpp/v1` folder missing from the 10.9 SDK folder, which the compiler expects. I'm going to write a small cmake patch that attempts to fix this...

Patch is done, just don't know where to put it... Here is it... https://gist.github.com/tresf/9e5043d431e5f9089a9b2f00c0a6fb64

Reply:
>  Is it still needed? Should I apply it onto the PR?

I think having standard C++ features will help us in the future, yes.

Reply:
>  I just pushed a commit to see what gh-actions does with the malloc/free version.

Cool however we can't reliably test my patch's effectiveness now that this PR has been updated.